Ryujinx.Ui.Common/App/ApplicationLibrary.cs

```
@@ -526,6 +526,176 @@ internal ApplicationMetadata LoadAndSaveMetaData(string titleId, Action<Applicat
+                            }
+
+                            try
+                            {
+                                file.Seek(24, SeekOrigin.Begin);
+
+                                int assetOffset = reader.ReadInt32();
+
+                                if (Encoding.ASCII.GetString(Read(assetOffset, 4)) == "ASET")
+                                {
+                                    byte[] iconSectionInfo = Read(assetOffset + 8, 0x10);
+
+                                    long iconOffset = BitConverter.ToInt64(iconSectionInfo, 0);
+                                    long iconSize = BitConverter.ToInt64(iconSectionInfo, 8);
+
+                                    // Reads and stores game icon as byte array
+                                    applicationIcon = Read(assetOffset + iconOffset, (int)iconSize);
+                                }
+                                else
+                                {
+                                    applicationIcon = _nroIcon;
+                                }
+                            }
+                            catch
+                            {
+                                Logger.Warning?.Print(LogClass.Application,
+                                    $"The file encountered was not of a valid type. Errored File: {applicationPath}");
+                            }
+                        }
+                        else if (Path.GetExtension(applicationPath).ToLower() == ".nca")
+                        {
+                            applicationIcon = _ncaIcon;
+                        }
+                        // If its an NSO we just set defaults
+                        else if (Path.GetExtension(applicationPath).ToLower() == ".nso")
+                        {
+                            applicationIcon = _nsoIcon;
+                        }
+                    }
+                }
+            }
+            catch { }
```

Do we really want to silence errors here?